testutil/checkers_test.go

```diff
@@ -0,0 +1,72 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	. "gopkg.in/check.v1"
+	"testing"
+)
+
+func Test2(t *testing.T) {
+	TestingT(t)
+}
+
+type CheckersSuite struct{}
+
+var _ = Suite(&CheckersSuite{})
+
+func (s *CheckersSuite) TestUnsupportedTypes(c *C) {
+	c.ExpectFailure("haystack is of unsupported type int")
+	c.Assert(5, Contains, "foo")
+}
+
+func (s *CheckersSuite) TestContainsVerifiesTypes(c *C) {
+	c.ExpectFailure("haystack contains items of type int but needle is a string")
```

Comment:
Ah, thanks. I've noticed that ExpectFailure is a bit buggy as it accepts _any_ string but I didn't know about the separate assert. I'll make that change.